gobblin-compaction/src/main/java/org/apache/gobblin/compaction/action/CompactionCompleteFileOperationAction.java

```diff
@@ -144,28 +147,32 @@ public void onCompactionJobComplete(FileSystemDataset dataset) throws IOExceptio
         Counter counter = job.getCounters().findCounter(RecordKeyMapperBase.EVENT_COUNTER.RECORD_COUNT);
         newTotalRecords = counter.getValue();
       }
+      final Path finalDstPath = dstPath;
       goodPaths.stream().forEach(p -> {
         String fileName = p.getName();
-        outputFiles.add(new Path(dstPath, fileName));
+        outputFiles.add(new Path(finalDstPath, fileName));
       });
       this.configurator.setDstNewFiles(outputFiles);
 
       State compactState = helper.loadState(new Path(result.getDstAbsoluteDir()));
-      if(executeCount!=0) {
-        compactState.setProp(CompactionSlaEventHelper.RECORD_COUNT_TOTAL + Long.toString(executeCount), Long.toString(helper.readRecordCount(new Path(result.getDstAbsoluteDir()))));
-        compactState.setProp(CompactionSlaEventHelper.EXEC_COUNT_TOTAL + Long.toString(executeCount), Long.toString(executeCount));
-        compactState.setProp("DuplicateRecordCount" + Long.toString(executeCount), compactState.getProp("DuplicateRecordCount", "null"));
+      if (executeCount != 0) {
+        compactState.setProp(CompactionSlaEventHelper.RECORD_COUNT_TOTAL + Long.toString(executeCount),
```

Comment:
Can we rename compactState to compactionState?

Reply:
Will address